src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,30 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  setSelectedTodo: (selectedTodo: Todo | null) => void;
+  selectedTodo: Todo;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(selectedTodo.userId).then(setSelectedUser);
+  }, [selectedTodo]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!selectedUser ? (
         <Loader />
```

Comment:
For loading you should create a separate state, as in case of any error the user sees the infinite loader